### PR TITLE
feat(parsing): More realistic locales in generated test data, cleaner line parsing logic.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Locales can have commas and "" in them, add a bunch of test locales
to ensure the parser can handle these.

While I was in there, I cleaned up the line parsing logic.

@jrgm - r?

### CODE REVIEW

File: lib/readers/csv.js [hunk 1]
```diff
@@ -189,9 +191,21 @@ class CSVReader extends EventEmitter {
...
     return new Date(time);
   }
+
+  _normalizeCSVLocale (locale) {
+    return locale && locale.trim();
+  }
+
+  _normalizeCSVEmail (email) {
+    return email && email.toLowerCase();
+  }
```
Comment (jrgm) [inline lib/readers/csv.js]: Hmm. Didn't I have to switch from pulling normalizedEmail out of the database, to pulling email? When I ran with fxa.accounts.normalizedEmail, I had lots of spurious changed accounts due to Foo@bar.com vs. foo@bar.com.

**Comment by jrgm** [comment]
So, to get this to work, I switched to the line being 'uid,b64-email,b64-locale,epochMs'. I'm not sure how I can reconcile those changes with these.

**Comment by jrgm** [comment]
I pushed up my changes as https://github.com/mozilla/fxa-reconcile-with-salesforce/tree/fix-mysql-email-locale-base64

**Comment by shane-tomlinson** [comment]
Let's go with your PR, closing this.

File: lib/readers/csv.js [hunk 2]
```diff
@@ -170,6 +148,30 @@ class CSVReader extends EventEmitter {
...
+    const [ uid, email ] = this._splitLineBuffer(salesforceLine);
+    return {
+      email,
+      normalizedEmail: this._normalizeCSVEmail(email),
+      normalizedUid: this._normalizeCSVUid(uid),
+      timestamp,
+      uid,
+    };
+  }
+
```
Comment (jrgm) [inline lib/readers/csv.js]: Actually, this would be good to revisit. Returning an object it cleaner code-wise than a variable length array.

Comment (shane-tomlinson) [reply lib/readers/csv.js]: I'll open a PR with these changes.

**Pull request closed without merging**